app/earn/[...vault]/form/form-button/form-button.tsx

```diff
@@ -43,10 +45,14 @@ const FormButton: React.FC<Props> = ({ vaultChain, isLoading, disabled, ...restP
     void submit(formAction)
   }, [insured, setInsured, status, isOnDifferentChain, formAction, connect, setCurrentChain, submit, vaultChain.id])
 
+  const shouldBeAblToSubmit = status === WalletStatus.CONNECTED
```

Comment:
```suggestion
  const shouldBeAbleToSubmit = status === WalletStatus.CONNECTED
```

Reply:
or even
```suggestion
  const isAbleToSubmit = status === WalletStatus.CONNECTED
```

Reply:
This not check if user able to submit, rather check should we check "is he able to submit". So I will keep it as "shouldBeAbleToSubmit"